drivers/dma/dma_nxp_edma.c

```diff
@@ -583,19 +578,36 @@ static int edma_get_attribute(const struct device *dev, uint32_t type, uint32_t
 
 static bool edma_channel_filter(const struct device *dev, int chan_id, void *param)
 {
-	int *requested_channel;
+	struct edma_channel *chan;
 
 	if (!param) {
 		return false;
 	}
 
-	requested_channel = param;
+	if (*(int *)param != chan_id) {
+		return false;
+	}
 
-	if (*requested_channel == chan_id && lookup_channel(dev, chan_id)) {
-		return true;
+	chan = lookup_channel(dev, chan_id);
+	if (!chan) {
+		return false;
 	}
 
-	return false;
+	irq_enable(chan->irq);
+
+	return true;
+}
+
+static void edma_channel_release(const struct device *dev, uint32_t chan_id)
+{
+	struct edma_channel *chan = lookup_channel(dev, chan_id);
+	if (!chan) {
+		return;
+	}
+
+#ifdef CONFIG_NXP_IRQSTEER
```

Comment:
If you disable the irq only for CONFIG_NXP_IRQSTEER, in other cases what happens?
Since, above, in `edma_channel_filter()` you enable the irq for "all".

Reply:
This is somewhat the way we used to do things. We used to call `irq_enable()` in `dma_config()`, resulting in `irq_enable` being called multiple times with no `irq_disable()` so the approach should be fine. There's no reference counting for non-irqstr-based SoCs.